### PR TITLE
Raise informative errors on missing non-python dependencies when invoked

### LINKED ISSUE #137: Provide information to users if non-python binaries like julia / buildcell not found in environment
Could be better to raise informative actions users can take when one wants to use any functionality that needs non python dependencies that are not directly installed when using pip install?

Maybe some function like this can do it? 

```python
import shutil

def check_binary_exists(binary_name):
    binary_path = shutil.which(binary_name)
    if binary_path:
        print(f"{binary_name} is installed at {binary_path}")
        return True
    else:
        print(f"{binary_name} is not installed or not in PATH, Please refer documentation to know how to fix this") # Maybe we can also point to URL directly once docs are deployed.
        return False
```

### PR DESCRIPTION
**Pull request by naik-aakash**
Closes #137 

## Changes

1. Added requires where possible
2. Added try except block to check lammps compilation
3. Minor update to docker file that only affects lammps usage directly via command line

### CODE REVIEW

**Comment by JaGeo** [comment]
@QuantumChemist can you please step in? Thanks!

File: src/autoplex/fitting/common/utils.py [hunk 1]
```diff
@@ -246,6 +249,12 @@ def gap_fitting(
     }
 
 
+@requires(
+    which("julia"),
+    "J-ACE fitting requires the executable 'julia' and ACEPotentials.jl library to be in PATH. "
+    "Please follow the instructions in the README (https://autoatml.github.io/autoplex/user/index.html) to install "
```
[the remark this answers is not in the document]
Comment (naik-aakash) [reply src/autoplex/fitting/common/utils.py:255]: Yup, will change it to documentation

[the remark this answers is not in the document]
Comment (QuantumChemist) [reply src/autoplex/fitting/common/utils.py:255]: Sounds good :)

[the remark this answers is not in the document]
Comment (naik-aakash) [reply src/autoplex/fitting/common/utils.py]: Done

**Review by QuantumChemist — APPROVED** [review]

**Pull request merged**